### DISCUSSION TITLE
Match MongoDB

### DISCUSSION THREAD

**Discussion by thauealfredo** [frontendbr/forum#1974] [Perguntas e Respostas]
Como eu consigo fazer esse userInfo.name ser apenas na primeira posição do userInfo ? Não posso trazer apenas a primeira posição do array porque uso as outras tbm. Adicionei esse and como uma forma de validar para o primeiro index do array, mas n consigo. Ja tentei com $cond, indesOfArray, indexOfByte
![image](https://user-images.githubusercontent.com/50925608/104517931-acc63d00-55d5-11eb-856f-17b36feedb5a.png)

**Comment by portothree** [comment]
Antes do `$match` faz um [$unwind](https://docs.mongodb.com/manual/reference/operator/aggregation/unwind/) para desconstruir a array em varios documentos

```
{
 "$unwind": "userInfo.name",
 "$match": {
    "customerId": customerId,
    "userInfo.name": new RegExp(query, "i")
  }
}
```
